utils/retrieveAsset.ts

```diff
@@ -9,42 +8,58 @@ import { retrieveAssetsFromStarkscan } from "./starkscan/retrieveAssetsFromStark
 import { execWithRateLimit } from "./execWithRateLimit";
 
 type RetrieveAssetsParameters = {
-  starknetNetwork: NetworkName;
+  network: NetworkName;
   contractAddress: string;
   ownerAddress: string;
   customApi?: CustomApi;
   address?: string;
 };
 
 export const retrieveAssets = async ({
-  starknetNetwork,
+  network,
   contractAddress,
   ownerAddress,
   customApi,
   address,
 }: RetrieveAssetsParameters) => {
-  if (!customApi || !customApi.apiUri)
+  // Check if network is Ethereum and handle accordingly
+  if (network === "ethereum" || network === "mainnet") {
+    const result = await fetchFromEtherscan(contractAddress, ownerAddress);
```

Comment:
"mainnet" is actually starknet's mainnet, so please remove it from the condition. Only fetch from etherscan if the network is ethereum-mainnet